openmeter/ent/schema/billing.go

```diff
@@ -315,76 +333,128 @@ type BillingInvoice struct {
 
 func (BillingInvoice) Mixin() []ent.Mixin {
 	return []ent.Mixin{
+		// This cannot be a resource mixin as the invoice doesn't have a name field
 		entutils.IDMixin{},
 		entutils.NamespaceMixin{},
-		entutils.TimeMixin{},
 		entutils.MetadataAnnotationsMixin{},
+		entutils.TimeMixin{},
+		entutils.CustomerAddressMixin{
+			FieldPrefix: "supplier",
+		},
+		entutils.CustomerAddressMixin{
+			FieldPrefix: "customer",
+		},
 	}
 }
 
 func (BillingInvoice) Fields() []ent.Field {
 	return []ent.Field{
-		// Invoice number
-		field.String("series").
+		// Customer/supplier
+		field.String("supplier_name").
+			NotEmpty(),
+
+		field.String("supplier_tax_code").
+			Optional().
+			Nillable(),
+
+		field.String("customer_name").
```

Comment:
do we need both supplier and customer name?
Customer name is also confusing as we have customer entity.

Reply:
By definition the invoice must clone the state of the customer and the supplier latest at the time the invoice is sent to the customer, so that it contains the values at the time of invoice creation and not at current time.

This is why we are cloning the objects.

If this wouldn't be the case a change in the customer entity would be retroactively refracted in the already sent invoices.